### PR TITLE
Docs: Argument Clinic: Improve 'How to write a custom converter'

### PR DESCRIPTION
**Pull request by erlend-aasland**
- ~~Move the CConverter class reference to the reference~~ (reverted; will make a separate PR for this)
- Omit unneccesary wording and sentences
- Don't mention implementation details (no digression, explanation)

----
:books: Documentation preview :books:: https://cpython-previews--107328.org.readthedocs.build/

### CODE REVIEW

**Comment by erlend-aasland** [comment]
Follow-up PRs: chose a better example (the current example is not even valid; there is no `Py_ssize_t` converter in zlibmodule.c. Also, it is not the simplest possible converter, as the current howto states. We also need to sync the CConverter class reference so it's up to date with the clinic source code.

**Comment by erlend-aasland** [comment]
FTR, I reverted moving the CConverter reference to the Reference section, to make this easier to review.

File: Doc/howto/clinic.rst [hunk 1]
```diff
@@ -1343,35 +1343,37 @@ state.  Example from the ``setattro`` slot method in
...
-
-You shouldn't subclass :py:meth:`!CConverter.__init__`.  Instead, you should
-write a :py:meth:`!converter_init` function. :py:meth:`!converter_init`
-always accepts a *self* parameter; after that, all additional
-parameters *must* be keyword-only.  Any arguments passed in to
-the converter in Argument Clinic will be passed along to your
-:py:meth:`!converter_init`.
+Your converter class should be named :samp:`{ConverterName}_converter`.
+By following this convention, your converter class will be automatically
+registered with Argument Clinic, with its *converter name* being be the name of
```
Comment (ezio-melotti) [inline Doc/howto/clinic.rst]:
```suggestion
registered with Argument Clinic, with its *converter name* being the name of
```

**Review by ezio-melotti — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Thanks for the review, Ezio!

**Pull request merged**